drivers/dma/dma_stm32_v3.c

```diff
@@ -0,0 +1,690 @@
+		LOG_ERR("Decrement mode not supported for source address adjustment");
+		return -ENOTSUP;
+	default:
+		LOG_ERR("Invalid source increment mode: %d", increment);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int dma_stm32_get_dest_inc_mode(enum dma_addr_adj increment, uint32_t *ll_increment)
+{
+	switch (increment) {
+	case DMA_ADDR_ADJ_INCREMENT:
+		*ll_increment = LL_DMA_DEST_INCREMENT;
+		break;
+	case DMA_ADDR_ADJ_NO_CHANGE:
+		*ll_increment = LL_DMA_DEST_FIXED;
+		break;
+	case DMA_ADDR_ADJ_DECREMENT:
+		LOG_ERR("Decrement mode not supported for destination address adjustment");
+		return -ENOTSUP;
+	default:
+		LOG_ERR("Invalid destination increment mode: %d", increment);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static void dma_stm32_get_src_data_width(uint32_t size, uint32_t *ll_src_data_width)
+{
+	static const uint32_t table_src_size[] = {
+		LL_DMA_SRC_DATAWIDTH_BYTE,
+		LL_DMA_SRC_DATAWIDTH_HALFWORD,
+		LL_DMA_SRC_DATAWIDTH_WORD,
+		LL_DMA_SRC_DATAWIDTH_DOUBLEWORD,
+	};
+
+	*ll_src_data_width = table_src_size[LOG2(size)];
+}
+
+static void dma_stm32_get_dest_data_width(uint32_t size, uint32_t *ll_dest_data_width)
+{
+	static const uint32_t table_dst_size[] = {
+		LL_DMA_DEST_DATAWIDTH_BYTE,
+		LL_DMA_DEST_DATAWIDTH_HALFWORD,
+		LL_DMA_DEST_DATAWIDTH_WORD,
+		LL_DMA_DEST_DATAWIDTH_DOUBLEWORD,
+	};
+
+	*ll_dest_data_width = table_dst_size[LOG2(size)];
+}
+
+static int dma_stm32_get_ll_direction(enum dma_channel_direction direction, uint32_t *ll_direction)
+{
+	switch (direction) {
+	case MEMORY_TO_MEMORY:
+		*ll_direction = LL_DMA_DIRECTION_MEMORY_TO_MEMORY;
+		break;
+	case MEMORY_TO_PERIPHERAL:
+		*ll_direction = LL_DMA_DIRECTION_MEMORY_TO_PERIPH;
+		break;
+	case PERIPHERAL_TO_MEMORY:
+		*ll_direction = LL_DMA_DIRECTION_PERIPH_TO_MEMORY;
+		break;
+	default:
+		LOG_ERR("Direction error. %d", direction);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int dma_stm32_get_priority(uint8_t priority, uint32_t *ll_priority)
+{
+	static const uint32_t table_priority[] = {
+		LL_DMA_LOW_PRIORITY_LOW_WEIGHT,
+		LL_DMA_LOW_PRIORITY_MID_WEIGHT,
+		LL_DMA_LOW_PRIORITY_HIGH_WEIGHT,
+		LL_DMA_HIGH_PRIORITY,
+	};
+
+	if (priority >= ARRAY_SIZE(table_priority)) {
+		LOG_ERR("Priority error.");
+		return -EINVAL;
+	}
+
+	*ll_priority = table_priority[priority];
+	return 0;
+}
+
+static bool dma_stm32_is_tc_irq_active(DMA_TypeDef *dma, uint32_t channel)
+{
+	return LL_DMA_IsEnabledIT_TC(dma, channel) && LL_DMA_IsActiveFlag_TC(dma, channel);
+}
+
+static bool dma_stm32_is_ht_irq_active(DMA_TypeDef *dma, uint32_t channel)
+{
+	return LL_DMA_IsEnabledIT_HT(dma, channel) && LL_DMA_IsActiveFlag_HT(dma, channel);
+}
+
+static void dma_stm32_disable_it(DMA_TypeDef *dma, uint32_t channel)
+{
+	mem_addr_t dma_channel_ccr_reg = (mem_addr_t)&dma_stm32_get_channel_addr(dma, channel)->CCR;
+
+	sys_clear_bits(dma_channel_ccr_reg, DMA_CCR_TCIE | DMA_CCR_HTIE | DMA_CCR_USEIE |
+						    DMA_CCR_ULEIE | DMA_CCR_DTEIE | DMA_CCR_SUSPIE);
```

Comment:
Indentation, remove a tab:
```suggestion
	sys_clear_bits(dma_channel_ccr_reg, DMA_CCR_TCIE | DMA_CCR_HTIE | DMA_CCR_USEIE |
					    DMA_CCR_ULEIE | DMA_CCR_DTEIE | DMA_CCR_SUSPIE);